sphinx_immaterial/kbd_keys.py

```diff
@@ -0,0 +1,112 @@
+from typing import List, Tuple
+from docutils import nodes
+from sphinx.application import Sphinx
+from sphinx.config import Config
+from sphinx.writers.html import HTMLTranslator
+from sphinx.util.logging import getLogger
+
+LOGGER = getLogger(__name__)
+
+try:
+    import pymdownx.keymap_db as keys_db
+except ImportError:
+    LOGGER.info(
+        "Could not import `keymap_db` module from `pymdownx` package.\n    "
+        "Please ensure `pymdown-extensions` is installed.\n    "
+        "The `:keys:` role has no default key map."
+    )
+    keys_db = None
+
+
+class spanNode(nodes.container):
+    pass
+
+
+def visit_span(self: HTMLTranslator, node: spanNode):
+    self.body.append("<span")
+    if "classes" in node and node.get("classes"):
+        self.body.append(
+            f' class="{" ".join([self.attval(cls) for cls in node["classes"]])}"'
+        )
+    if "text" in node:
+        self.body.append(">" + node["text"])
+    else:
+        self.body.append(">")
+
+
+def depart_span(self: HTMLTranslator, node: spanNode):
+    self.body.append("</span>")
+
+
+class kbdNode(nodes.container):
```

Comment:
I think you could just have a single `kbd` node that inherits from nodes.Text and holds the unparsed text.  Then in visit_kbd you can put all of the logic for parsing it and outputting the spans.  You wouldn't need the extra `spanNode` that way.  (The idea is that the docutils nodes would better reflect the semantic structure of the document, and you would leave the HTML-specific stuff to the HTML translation step.)

Reply:
I looked into how the original kbd role works, and it looks like the `nodes.container` is used. IIRC,  when `visit_container()` is called, the tag is changed (from div to span or kbd) based on the node's context/flags.

Reply:
Yeah, you could still create a custom `kbd` node type but just override the existing `kbd` role.

Reply:
`kbd_node` is now the only new node introduced by kbd_keys.py. I had to inherit from `nodes.TextElement` because `nodes.Text` was too generic.

As for latex, I just have the key's displayed text inserted as plain text (with separators when applicable).